process/factory/metachain/preProcessorsContainerFactory_test.go

```diff
@@ -0,0 +1,269 @@
+
+	assert.Equal(t, process.ErrNilShardCoordinator, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilStore(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		nil,
+		&mock.MarshalizerMock{},
+		&mock.HasherMock{},
+		mock.NewPoolsHolderMock(),
+		&mock.AccountsStub{},
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Equal(t, process.ErrNilStore, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilMarshalizer(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.ChainStorerMock{},
+		nil,
+		&mock.HasherMock{},
+		mock.NewPoolsHolderMock(),
+		&mock.AccountsStub{},
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Equal(t, process.ErrNilMarshalizer, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilHasher(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.ChainStorerMock{},
+		&mock.MarshalizerMock{},
+		nil,
+		mock.NewPoolsHolderMock(),
+		&mock.AccountsStub{},
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Equal(t, process.ErrNilHasher, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilDataPool(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.ChainStorerMock{},
+		&mock.MarshalizerMock{},
+		&mock.HasherMock{},
+		nil,
+		&mock.AccountsStub{},
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Equal(t, process.ErrNilDataPoolHolder, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilAccounts(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.ChainStorerMock{},
+		&mock.MarshalizerMock{},
+		&mock.HasherMock{},
+		mock.NewPoolsHolderMock(),
+		nil,
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Equal(t, process.ErrNilAccountsAdapter, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilFeeHandler(t *testing.T) {
+	t.Parallel()
+
+	ppcm, err := metachain.NewPreProcessorsContainerFactory(
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.ChainStorerMock{},
+		&mock.MarshalizerMock{},
+		&mock.HasherMock{},
+		mock.NewPoolsHolderMock(),
+		&mock.AccountsStub{},
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		nil,
+	)
+
+	assert.Equal(t, process.ErrNilEconomicsFeeHandler, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilTxProcessor(t *testing.T) {
```

Comment:
maybe rearrange the tests in order to check nil parameters in order ? (tip: select a test name and ctrl+shift+up)

Reply:
not critical.